### ISSUE TITLE
URI redirect issue (yes: Again)

### ISSUE THREAD

**Issue by TeamRainless** [shyambhat/InstagramKit#244]
I'm really willing to rewrite the part of the ReadMe that deals with authentication because I've read all the other issues on this page... and the Stack Overflow posts... and I still don't get it. 

First off I'm using the "dev" branch.

In my Info.Plist I have:

`InstagramAppRedirectURL: net.myapp.app://authURL`

And also on instagram I have:

`Redirect URI: net.myapp.app://authURL`

It's always identical... it never works. (and I've tried [myclientID]://authorize and a bunch of other things.)

I get the error "Error: The page could not be loaded If you have cookies..."

Here's my code:

```
- (void)viewDidLoad {
    [super viewDidLoad];
    
    NSLog(@"Do we see the light of day?");
    InstagramKitLoginScope scope = InstagramKitLoginScopeRelationships | InstagramKitLoginScopeComments | InstagramKitLoginScopeLikes;
    NSURL *authURL = [[InstagramEngine sharedEngine] authorizationURLForScope:scope];
    [self.webView loadRequest:[NSURLRequest requestWithURL:authURL]];
    
   
}

- (BOOL)webView:(UIWebView *)webView shouldStartLoadWithRequest:(NSURLRequest *)request navigationType:(UIWebViewNavigationType)navigationType
{
    NSLog(@"Did instagram make it here?");
    NSError *error;
    if ([[InstagramEngine sharedEngine] receivedValidAccessTokenFromURL:request.URL error:&error]) {
        // success!
        NSLog(@"Instagram login successful");
    }
    return YES;
}
```

We do "see the light of day".  I make it to the login screen... I login... but it never makes it to "Did instagram make it here?"

Any help would be appreciated.

**Comment by TeamRainless** [comment]
Ok.  Found the answer.  You need to set the WebView delegate in "ViewWillAppear"... otherwise it will never make it to "shouldStartLoadWithRequest"

```
-(void)viewWillAppear:(BOOL)animated
{
    _webView.delegate = self;
}
```

**Comment by NirmitDagly** [comment]
Hello,
I am developing iOS application and used this repo for integration of Instagram into project. I have followed the readme and also googled about how to open the app after successful login into Instagram and generating token. But In my case, I am not able to redirect into application. 

Here is what happening..

When I am trying to define redirect_uri to Instagram, it is always telling me to register with http:// or https:// only and not by my app name just like xyz:// which is defined as per the custom URL scheme.

Now, when I have set the redirect_uri based on the http://xyz and configure the same into custom URL scheme in my XCode project, this won't allow the app opening automatically and redirects to the web page in safari.

So, I would like to know the possible workaround for this. Any help will be appreciated.
